packages/Modal/src/index.tsx

```diff
@@ -66,25 +66,26 @@ const ModalComponent = forwardRef<'div', ModalProps>(
     },
     ref
   ) => {
-    const closeModal = () => {
+    const prevHide = useMemo(() => state.hide, [state])
+
+    state.hide = useCallback(() => {
```

Comment:
maybe it's better to put that directly in the `useModalState` ?

Reply:
The custom onClose [is passed as a prop](https://github.com/WTTJ/welcome-ui/blob/fc1db748468c974130d7f7224e2f37f4b246b695/packages/Modal/src/index.tsx#:~:text=export%20interface%20ModalOptions,%2C%20ModalOptions%3E)
Should we change it to a hook option and remove it form the modalProps ?

Reply:
I tend to agree with Mick, monkey patching the state & mutating the state isn't great imo